base/MAPL_ObsUtil.F90

```diff
@@ -0,0 +1,544 @@
+          call get_var_from_name_w_group (var_name_time, time_loc_R8, filename, _RC)
+          time(1:nlon,jx+1:jx+nlat) = time_loc_R8(1:nlon,1:nlat)
+          deallocate(time_loc_R8)
+       end if
+
+       if (present(var_name_lon).AND.present(lon)) then
+          allocate (lon_loc(nlon, nlat))
+          call get_var_from_name_w_group (var_name_lon, lon_loc, filename, _RC)
+          lon(1:nlon,jx+1:jx+nlat) = lon_loc(1:nlon,1:nlat)
+          deallocate(lon_loc)
+       end if
+
+       if (present(var_name_lat).AND.present(lat)) then
+          allocate (lat_loc(nlon, nlat))
+          call get_var_from_name_w_group (var_name_lat, lat_loc, filename, _RC)
+          lat(1:nlon,jx+1:jx+nlat) = lat_loc(1:nlon,1:nlat)
+          deallocate(lat_loc)
+       end if
+
+       jx = jx + nlat
+
+    end do
+
+    _RETURN(_SUCCESS)
+  end subroutine read_M_files_4_swath
+
+
+  !
+  !-- caveat: note call this subr. on head node
+  !           because of (bash ls) command therein
+  !
+  function get_filename_from_template_use_index (obsfile_start_time, obsfile_interval, &
+       f_index, file_template, rc) result(filename)
+    use Plain_netCDF_Time, only : ESMF_time_to_two_integer
+    use MAPL_StringTemplate, only : fill_grads_template    
+    character(len=ESMF_MAXSTR) :: filename
+    type(ESMF_Time), intent(in) :: obsfile_start_time
+    type(ESMF_TimeInterval), intent(in) :: obsfile_interval
+    character(len=*), intent(in) :: file_template
+    integer, intent(in) :: f_index
+    integer, optional, intent(out) :: rc
+
+    integer :: itime(2)
+    integer :: nymd, nhms
+    integer :: status
+    real(ESMF_KIND_R8) :: dT0_s
+    real(ESMF_KIND_R8) :: s
+    type(ESMF_TimeInterval) :: dT
+    type(ESMF_Time) :: time
+    integer :: i, j
+
+    character(len=ESMF_MAXSTR) :: file_template_left
+    character(len=ESMF_MAXSTR) :: file_template_right
+    character(len=ESMF_MAXSTR) :: filename_left
+    character(len=ESMF_MAXSTR) :: filename_full
+    character(len=ESMF_MAXSTR) :: filename2
+    character(len=ESMF_MAXSTR) :: cmd
+
+    call ESMF_TimeIntervalGet(obsfile_interval, s_r8=dT0_s, rc=status)
+    s = dT0_s * f_index
+    call ESMF_TimeIntervalSet(dT, s_r8=s, rc=status)
+    time = obsfile_start_time + dT
+
+    call ESMF_time_to_two_integer(time, itime, _RC)
+    nymd = itime(1)
+    nhms = itime(2)
+
+
+       j= index(file_template, '*')
+       if (j>0) then
+          ! wild char exist
+          !!print*, 'pos of * in template =', j
+          file_template_left = file_template(1:j-1)
+          call fill_grads_template ( filename_left, file_template_left, &
+               experiment_id='', nymd=nymd, nhms=nhms, _RC )
+          filename= trim(filename_left)//trim(file_template(j:))
+          cmd="bash -c 'ls "//trim(filename)//"' &> zzz_MAPL"
+          CALL execute_command_line(trim(cmd))
+          open(7213, file='zzz_MAPL', status='unknown')
+          read(7213, '(a)') filename
```

Comment:
Do not use hardcoded units.  (Indeed, I don't think units over 100 are even guaranteed to work by the standard.)

Instead use the `newunit` argument on the open statement.  See me if you need assistance with that.

Reply:
Thank you, you get where I was fiddling, ending up with a prime number in hope not to conflict with used file unit.  Thanks for pointing out the functionality in Fortran 2008. I find the reference on web and the example in MAPL.